main.py

```diff
@@ -534,52 +534,80 @@ async def list_result_directory() -> list[int]:
 
 
 @app.get("/dataset/master/")
-async def get_master_dataset(key: str) -> tuple[float, Union[int, float, list]]:
-    """Returns the current timestamp and the dataset broadcast to artiq master.
+async def get_master_dataset(key: str) -> Union[int, float, list]:
+    """Returns the dataset broadcast to artiq master.
 
     Args:
         key: The key of the target dataset.
 
     Returns:
-        If the dataset is not initialized or does not exist, it returns (-1, ()).
+        If the dataset is not initialized or does not exist, it returns an empty tuple.
     """
-    timestamp, array = dataset_tracker.get(key)
-    if isinstance(array, np.ndarray):
-        array = array.tolist()
-    return timestamp, array
+    _, data = dataset_tracker.get(key)
+    if isinstance(data, np.ndarray):
+        data = data.tolist()
+    return data
 
 
-@app.get("/dataset/master/list/")
-async def list_dataset() -> tuple[str, ...]:
-    """Returns the list of datasets available in artiq master."""
-    return dataset_tracker.datasets()
+@app.websocket("/dataset/master/list/")
+async def list_dataset(websocket: WebSocket):
+    """Sends the list of datasets available in artiq master whenever it is modified.
+    
+    After accepted, it sends the current dataset list immediately.
+    Then, it sends the dataset list every time it is modified.
 
+    Args:
+        websocket: The web socket object.
+    """
+    await websocket.accept()
+    try:
+        datasets = dataset_tracker.datasets()
+        await websocket.send_json(datasets)
```

Comment:
Since `datasets` is not used in any other places and the method name `datasets()` is clear enough, I think you can just write it in a single line:
```python
await websocket.send_json(dataset_tracker.datasets())`
```